nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/PublishGCPubSubLite.java

```diff
@@ -0,0 +1,330 @@
+    public void onScheduled(final ProcessContext context) {
+        try {
+            if (publisher == null) {
+                publisher = getPublisher(context);
+            }
+        } catch (final Exception e) {
+            getLogger().error("Failed to create Google Cloud PubSub Lite Publisher", e);
+            throw new ProcessException(e);
+        }
+        try {
+            publisher.startAsync().awaitRunning();
+        } catch (final Exception e) {
+            getLogger().error("Failed to create Google Cloud PubSub Lite Publisher", publisher.failureCause());
+            throw new ProcessException(e);
+        }
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final int flowFileCount = context.getProperty(BATCH_SIZE).asInteger();
+        final List<FlowFile> flowFiles = session.get(flowFileCount);
+
+        if (flowFiles.isEmpty()) {
+            context.yield();
+            return;
+        }
+
+        if (publisher == null) {
+            getLogger().error("Google Cloud PubSub Lite Publisher was not properly created. Yielding the processor...");
+            context.yield();
+            return;
+        }
+
+        if(!publisher.isRunning()) {
+            getLogger().error("Google Cloud PubSub Lite Publisher is not running. Yielding the processor...", publisher.failureCause());
+            throw new ProcessException(publisher.failureCause());
+        }
+
+        final long startNanos = System.nanoTime();
+        final List<FlowFile> successfulFlowFiles = new ArrayList<>();
+        final String topicName = context.getProperty(TOPIC_NAME).evaluateAttributeExpressions().getValue();
+        final List<ApiFuture<String>> futures = new ArrayList<>();
+
+        try {
+            for (FlowFile flowFile : flowFiles) {
+                final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                session.exportTo(flowFile, baos);
+                final ByteString flowFileContent = ByteString.copyFrom(baos.toByteArray());
+                final String orderingKey = context.getProperty(ORDERING_KEY).evaluateAttributeExpressions(flowFile).getValue();
+
+                PubsubMessage message= null;
+                if (orderingKey != null) {
+                    message = PubsubMessage.newBuilder().setData(flowFileContent)
+                            .setOrderingKey(orderingKey)
+                            .setPublishTime(Timestamp.newBuilder().build())
+                            .putAllAttributes(getDynamicAttributesMap(context, flowFile))
+                            .build();
+                } else {
+                    message = PubsubMessage.newBuilder().setData(flowFileContent)
+                            .setPublishTime(Timestamp.newBuilder().build())
+                            .putAllAttributes(getDynamicAttributesMap(context, flowFile))
+                            .build();
+                }
```

Comment:
This can be streamlined a little by declaring the builder as a final local variable, and then adding the orderingKey if it's non-null.